types/json-patch-gen/index.d.ts

```diff
@@ -0,0 +1,21 @@
+// Type definitions for json-patch-gen 1.0
+// Project: https://github.com/gregsexton/json-patch-gen
+// Definitions by: Konstantin Rohde <https://github.com/RohdeK>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+// TypeScript Version: 2.2
+
+type PatchOperation = "replace" | "add" | "remove";
+
+interface JsonPatch {
+    op: PatchOperation;
+    path: string;
+    value: any;
+}
+
+type Diff = (obj1: object, obj2: object) => JsonPatch[];
+
+declare var diff: Diff;
+
+declare module "json-patch-gen" {
```

Comment:
Can you use the 2.x syntax to make a UMD module? Instead of `declare module...`, it's

```ts
export = diff
export as namespace diff;
```

Reply:
I extended this as a real namespace, so that e.g. the JsonPatch interface can be referred to from the outside.